homeassistant/components/modbus/climate.py

```diff
@@ -387,3 +398,17 @@ async def _async_read_register(
             return None
         self._attr_available = True
         return float(self._value)
+
+    def _GetMeTargetTempRegister(self) -> int:
+        """Get the target temp register for a specific HVAC mode."""
+        if isinstance(self._target_temperature_register, int):
```

Comment:
This is better done in the validator, always convert the parameter to an array...then we have less to do at runtime.